azure/durable_functions/models/DurableOrchestrationClient.py

```diff
@@ -74,7 +74,9 @@ async def start_new(self,
         if response[0] <= 202 and response[1]:
             return response[1]["id"]
         else:
-            return None
+            err_message = f"Failed to start orchestration: {orchestration_function_name}."\
+                "Did you make a typo?"
+            raise Exception(err_message)
```

Comment:
Why not just surface up the error message that came back from the extension? That's what we're doing for JavaScript ([link](https://github.com/Azure/azure-functions-durable-js/blob/6ed537cc96c83f037beb4aa5c54c8ea3c91c734e/src/durableorchestrationclient.ts#L695)).

Reply:
Ah, silly me. I was testing the Durable-JS behaviour and did not realize that it was throwing a _different_ kind of exception. I can definitely surface the same kind of exception, and will do. Thank you for pointing that out.

However, I do think we could benefit from some form of pretty-printing when it comes to raising these exceptions. This is how it looks for me on Durable-JS, I'm not surprised I didn't catch that it was a different kind of exception: there's a lot of noise.

![testo](https://user-images.githubusercontent.com/9623336/85638674-60b4aa80-b63b-11ea-9fed-b89b6d5cd150.PNG)

That's a less urgent discussion though :)

Reply:
I'm fine with that. In fact, it might not be hard to do. If you can detect that the response is JSON and has a "Message" field, then the error could simply be the Message text rather than "The operation failed with an unexpected status code..." followed by the full payload that includes this ugly call-stack.